torchrl/collectors/collectors.py

```diff
@@ -322,10 +322,76 @@ def __init__(
         self._tensordict.set(
             "step_count", torch.zeros(*self.env.batch_size, 1, dtype=torch.int)
         )
-        self._tensordict_out = TensorDict(
-            {},
-            batch_size=[*self.env.batch_size, self.frames_per_batch],
-            device=self.passing_device,
+
+        # TODO: perhaps check type of policy and raise TypeError if something
```

Comment:
we can remove this comment
Long term plan is to support any kind of policy so i'm happy with the state of the checks